src/sentry/rules/processing/processor.py

```diff
@@ -277,9 +314,15 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
 
         state = self.get_state()
         condition_list, filter_list = split_conditions_and_filters(rule.data.get("conditions", ()))
-
-        # Sort `condition_list` so that most expensive conditions run last.
-        condition_list.sort(key=lambda condition: is_condition_slow(condition))
+        fast_conditions, slow_conditions = self.group_conditions_by_speed(condition_list)
+        process_slow_conditions_later = features.has(
+            "organizations:process-slow-alerts", self.project.organization
+        )
+        if process_slow_conditions_later:
+            condition_list = fast_conditions
+        else:
+            fast_conditions.extend(slow_conditions)  # type: ignore[arg-type]
+            condition_list = fast_conditions
```

Comment:
nit: I think it might be good to just add the lists and include a comment
```suggestion
            # Condition_list is now ordered by speed
            condition_list = fast_conditions + slow_conditions
```